docs/tutorials/rerank/rerank_pipeline_with_Amazon_Rerank_model_on_Amazon_Bedrock.md

```diff
@@ -0,0 +1,622 @@
+  "description": "Test connector for Amazon Bedrock cross-encoder model",
+  "version": 1,
+  "protocol": "aws_sigv4",
+  "credential": {
+    "roleArn": "your_role_arn_which_allows_access_to_bedrock_model"
+  },
+  "parameters": {
+    "region": "your_bedrock_model_region_like_us-west-2",
+    "service_name": "bedrock"
+  },
+  "actions": [
+    {
+      "action_type": "predict",
+      "method": "POST",
+      "url": "https://bedrock-runtime.${parameters.region}.amazonaws.com/model/amazon.rerank-v1:0/invoke",
+      "headers": {
+        "x-amz-content-sha256": "required",
+        "content-type": "application/json"
+      },
+      "pre_process_function": """
+        def query_text = params.query_text;
+        def text_docs = params.text_docs;
+        def textDocsBuilder = new StringBuilder('[');
+        for (int i=0; i<text_docs.length; i++) {
+          textDocsBuilder.append('\"');
+          textDocsBuilder.append(text_docs[i]);
+          textDocsBuilder.append('\"');
+          if (i<text_docs.length - 1) {
+            textDocsBuilder.append(',');
+          }
+        }
+        textDocsBuilder.append(']');
+        def parameters = '{ \"query\": \"' + query_text + '\",  \"documents\": ' + textDocsBuilder.toString() + ' }';
+        return  '{\"parameters\": ' + parameters + '}';
+      """,
+      "request_body": "{ \"query\": \"${parameters.query}\", \"documents\": ${parameters.documents} }",
+      "post_process_function": """
+        if (params.result == null || params.result.length > 0) {
+          throw new IllegalArgumentException("Post process function input is empty.");
+        }
+        def outputs = params.results;
+        def relevance_scores = new Double[outputs.length];
+        for (int i=0; i<outputs.length; i++) {
+          def index = new BigDecimal(outputs[i].index.toString()).intValue();
+          relevance_scores[index] = outputs[i].relevance_score;
+        }
+        def resultBuilder = new StringBuilder('[');
+        for (int i=0; i<relevance_scores.length; i++) {
+          resultBuilder.append(' {\"name\": \"similarity\", \"data_type\": \"FLOAT32\", \"shape\": [1],');
+          resultBuilder.append('\"data\": [');
+          resultBuilder.append(relevance_scores[i]);
+          resultBuilder.append(']}');
+          if (i<outputs.length - 1) {
+            resultBuilder.append(',');
+          }
+        }
+        resultBuilder.append(']');
+        return resultBuilder.toString();
+      """
+    }
+  ]
+}
+```
+
+Use the connector ID from the response to register and deploy the model:
+```json
+POST /_plugins/_ml/models/_register?deploy=true
+{
+    "name": "Bedrock Cross-Encoder model",
+    "function_name": "remote",
+    "description": "test rerank model",
+    "connector_id": "your_connector_id"
+}
+```
+Note the model ID in the response; you'll use it in the following steps.
+
+Test the model by using the Predict API:
+```json
+POST _plugins/_ml/models/your_model_id/_predict
```

Comment:
Please make sure every step in the tutorial works. I tested and this step failed
```
{
  "error": {
    "root_cause": [
      {
        "type": "script_exception",
        "reason": "compile error",
        "script_stack": [
          "...     resultBuilder.append(' {\\\"name\\\": \\\"similarity ...",
          "                             ^---- HERE"
        ],
        "script": "\n        if (params.result == null || params.result.length > 0) {\n          throw new IllegalArgumentException(\"Post process function input is empty.\");\n        }\n        def outputs = params.results;\n        def relevance_scores = new Double[outputs.length];\n        for (int i=0; i<outputs.length; i++) {\n          def index = new BigDecimal(outputs[i].index.toString()).intValue();\n          relevance_scores[index] = outputs[i].relevance_score;\n        }\n        def resultBuilder = new StringBuilder('[');\n        for (int i=0; i<relevance_scores.length; i++) {\n          resultBuilder.append(' {\\\"name\\\": \\\"similarity\\\", \\\"data_type\\\": \\\"FLOAT32\\\", \\\"shape\\\": [1],');\n          resultBuilder.append('\\\"data\\\": [');\n          resultBuilder.append(relevance_scores[i]);\n          resultBuilder.append(']}');\n          if (i<outputs.length - 1) {\n            resultBuilder.append(',');\n          }\n        }\n        resultBuilder.append(']');\n        return resultBuilder.toString();\n      ",
        "lang": "painless",
        "position": {
          "offset": 598,
          "start": 573,
          "end": 623
        }
      }
    ],
    "type": "m_l_exception",
    "reason": "m_l_exception: Fail to execute PREDICT in aws connector",
    "caused_by": {
      "type": "script_exception",
      "reason": "compile error",
      "script_stack": [
        "...     resultBuilder.append(' {\\\"name\\\": \\\"similarity ...",
        "                             ^---- HERE"
      ],
      "script": "\n        if (params.result == null || params.result.length > 0) {\n          throw new IllegalArgumentException(\"Post process function input is empty.\");\n        }\n        def outputs = params.results;\n        def relevance_scores = new Double[outputs.length];\n        for (int i=0; i<outputs.length; i++) {\n          def index = new BigDecimal(outputs[i].index.toString()).intValue();\n          relevance_scores[index] = outputs[i].relevance_score;\n        }\n        def resultBuilder = new StringBuilder('[');\n        for (int i=0; i<relevance_scores.length; i++) {\n          resultBuilder.append(' {\\\"name\\\": \\\"similarity\\\", \\\"data_type\\\": \\\"FLOAT32\\\", \\\"shape\\\": [1],');\n          resultBuilder.append('\\\"data\\\": [');\n          resultBuilder.append(relevance_scores[i]);\n          resultBuilder.append(']}');\n          if (i<outputs.length - 1) {\n            resultBuilder.append(',');\n          }\n        }\n        resultBuilder.append(']');\n        return resultBuilder.toString();\n      ",
      "lang": "painless",
      "position": {
        "offset": 598,
        "start": 573,
        "end": 623
      },
      "caused_by": {
        "type": "illegal_argument_exception",
        "reason": "unexpected character [' {\\\"]. The only valid escape sequences in strings starting with ['] are [\\\\] and [\\'].",
        "caused_by": {
          "type": "lexer_no_viable_alt_exception",
          "reason": "lexer_no_viable_alt_exception: null"
        }
      }
    }
  },
  "status": 500
}
```

Reply:
I'm going to remove unnecessary escape character and test again on my side.

Reply:
fixed on 9482b472b2bc68bba61267c87f484967362e2c5b